src/test/java/com/commercetools/sync/taxcategories/TaxCategorySyncOptionsTest.java

```diff
@@ -0,0 +1,158 @@
+package com.commercetools.sync.taxcategories;
+
+import com.commercetools.sync.commons.utils.TriFunction;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.taxcategories.TaxCategory;
+import io.sphere.sdk.taxcategories.TaxCategoryDraft;
+import io.sphere.sdk.taxcategories.TaxCategoryDraftBuilder;
+import io.sphere.sdk.taxcategories.commands.updateactions.SetKey;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TaxCategorySyncOptionsTest {
+
+    private static SphereClient CTP_CLIENT = mock(SphereClient.class);
+
+    @Test
+    void applyBeforeUpdateCallback_WithNullCallback_ShouldReturnIdenticalList() {
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertThat(filteredList).as(" returned 'updateActions' should not be changed")
+            .isSameAs(updateActions);
+    }
+
+    @Test
+    void applyBeforeUpdateCallback_WithNullReturnCallback_ShouldReturnEmptyList() {
+        TriFunction<List<UpdateAction<TaxCategory>>, TaxCategoryDraft, TaxCategory, List<UpdateAction<TaxCategory>>>
+            beforeUpdateCallback = (updateActions, newCategory, oldCategory) -> null;
+        final TaxCategorySyncOptions taxCategorySyncOptions = TaxCategorySyncOptionsBuilder.of(CTP_CLIENT)
+            .beforeUpdateCallback(beforeUpdateCallback)
+            .build();
+        assertThat(taxCategorySyncOptions.getBeforeUpdateCallback()).isNotNull();
+
+        List<UpdateAction<TaxCategory>> updateActions = singletonList(SetKey.of("key"));
+        List<UpdateAction<TaxCategory>> filteredList = taxCategorySyncOptions
+            .applyBeforeUpdateCallBack(updateActions, mock(TaxCategoryDraft.class), mock(TaxCategory.class));
+
+        assertAll(
+            () -> assertThat(filteredList)
+                .as("returned 'updateActions' should not be equal to prepared ones")
+                .isNotEqualTo(updateActions),
+            () -> assertThat(filteredList).as("returned 'updateActions' should be empty").isEmpty()
```

Comment:
I think the empty check here is enough, since we know for a fact that it is not equal to the `updateActions` variable which is not empty.